utils/timeout.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef UTILS_TIMEOUT_H_
+#define UTILS_TIMEOUT_H_
+
+#include <pthread.h>
+#include <signal.h>
+#include <sys/time.h>
+#include "accelerator/common_core.h"
+#include "accelerator/config.h"
+#include "accelerator/errors.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file utils/timeout.h
+ */
+
+typedef struct {
+  const iota_config_t *iconf;
+  const iota_client_service_t *service;
+  ta_generate_address_res_t *res;
+} ta_generate_address_args_t;
+
+pthread_t ta_generate_address_thread_id;
+static const struct itimerval TA_TIMER_TIMEOUT = {.it_interval = {.tv_sec = 0, .tv_usec = 0},
+                                                  .it_value = {.tv_sec = 10, .tv_usec = 0}};
+static const struct itimerval TA_ZERO_TIMEOUT = {.it_interval = {.tv_sec = 0, .tv_usec = 0},
+                                                 .it_value = {.tv_sec = 0, .tv_usec = 0}};
+
+void setup_timeout();
+void remove_timeout();
+void timeout_handler(int signum);
+status_t ta_generate_address_thread(void *args);
```

Comment:
Shall we expose `ta_generate_address_thread` as public function?

Reply:
No, I'll make `setup_timeout`, `remove_timeout`, `timeout_handler`, and `ta_generate_address_therad` as static function.